httpbakery/discharge.go

```diff
@@ -273,3 +291,28 @@ func randomBytes(n int) ([]byte, error) {
 	}
 	return b, nil
 }
+
+// maybeBase64Encode encodes b as is if it's
+// OK to be passed as a URL form parameter,
+// or encoded as base64 otherwise.
+func maybeBase64Encode(b []byte) (s, s64 string) {
+	if utf8.Valid(b) {
```

Comment:
If it's valid ascii, it seems unnecessary to encode it as base64 - it takes more space as isn't as easy to debug.
I've chosen the same approach used by the macaroon encoding here.

I've also made the criteria a little more strict - it can't include ASCII control characters now.